gitops/PrivateRepositories.md

```diff
@@ -0,0 +1,111 @@
+  2. Invoke `fab install` from the same directory as the component.
+
+## SSH Keys
+
+While Personal Access Tokens offer a reliable way to read and clone Git repositories on behalf of a user, they can become unwieldy to use in practice. For an Azure DevOps scenario, imagine if a component's subcomponents were spread across multiple Organizations - a user supplying a Personal Access Token would have to provide one for every organization the component depends on. For example, given the following component.yaml:
+
+```
+name: episerver-cluster-hld
+type: component
+subcomponents:
+- name: private-component
+  type: component
+  source: https://dev.azure.com/contoso/cool-project/_git/private-component
+  method: git
+- name: awesome-component
+  type: component
+  source: https://dev.azure.com/fabrikam/awesome-project/_git/awesome-component
+  method: git
+```
+
+A corresponding access.yaml would like:
+
+```
+https://dev.azure.com/contoso/cool-project/_git/private-component: ENV_VAR_FOR_CONTOSO_PAT
+https://dev.azure.com/fabrikam/awesome-project/_git/awesome-component: ENV_VAR_FOR_FABRIKAM_PAT
+```
+
+And would require Personal Access Tokens for both the `contoso` organization and the `fabrikam` organization for a `fab install` to work as expected. Note that due to the recursive nature of components, a subcomponent can refer to other private git repositories as subcomponents as well. The subcomponent's root component.yaml can thus also provide an adjacent access.yaml, specifying the required Personal Access Tokens as environment variable names, and so on.
+
+Managing Personal Access Tokens can be utilized for simple components, but as you can imagine, the complexity increases dramatically as one adds more private components as subcomponents. As such, you may be interested in managing access to components via SSH Key Pairs, which many, if not all, Git providers provide as an authentication mechanism out of the box.
+
+To utilize SSH Key Pairs you must do the following:
+
+1. Utilizing a SSH toolkit ([we suggest OpenSSH](https://www.openssh.com/)), [generate an SSH Key Pair](https://www.ssh.com/ssh/keygen/).
+2. Upload the Public Key to Azure DevOps (the URL depends on your organization, but typically follows the pattern https://dev.azure.com/${organization}/_usersSettings/keys) or [GitHub](https://github.com/settings/keys).
+3. [Add the Private Key to the SSH Agent](https://www.ssh.com/ssh/add)
+
+Take note the following considerations:
+- For Azure Devops, this process must be repeated for every organization that your component depends on.
+- For Github, the user account to which the public key is added must have access to all the private repositories that your component depends on.
+
+With this setup, you can migrate away from Personal Access Tokens, and use a more secure Public/Private key approach to manage and maintain repository access, while still working with Fabrikate components from private repositories.
```

Comment:
We should include a disclaimer that although SSH Keys are now supported in Fabrikate verison 0.15.1, we cannot remove PATs entirely because the Bedrock CI/CD workflow still requires PRs to be made, and that strictly uses PATs.

Reply:
good point - i'll address the fact PATs can't be removed entirely - but they're no longer needed for configuring fabrikate to install components, right? the PAT that is utilized to make PRs/etc is setup elsewhere, i thought

Reply:
That's correct. They are no longer needed to install components as long as SSH Keys are involved. And yes, PATs used to make PRs do not involve Fabrikate in this setup.

Reply:
I think this is resolved via changes